src/Compilers/Core/Portable/MetadataReader/PEModule.cs

```diff
@@ -3638,6 +3639,32 @@ private ConstantValue GetConstantValueOrThrow(ConstantHandle handle)
 
             matchedName = null;
             return (FirstIndex: -1, SecondIndex: -1);
+
+            void ensureCaseInsensitiveDictionary()
+            {
+                if (_lazyCaseInsensitiveForwardedTypesToAssemblyIndexMap != null)
+                {
+                    return;
+                }
+
+                var caseInsensitiveMap = new Dictionary<string, OneOrMany<(int FirstIndex, int SecondIndex)>>(StringComparer.OrdinalIgnoreCase);
+
+                foreach (var (key, indexPair) in _lazyForwardedTypesToAssemblyIndexMap)
```

Comment:
>_lazyForwardedTypesToAssemblyIndexMap [](http://example.com/codeflow?start=49&length=37)

The majority of entries have zero entries (at least in Roslyn). Maybe it would be worthwhile to have an empty dictionary to use for those instead of allocating a new one?

Reply:
Fair observation.